submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/submitter/request/SubmissionRequestProcessor.kt

```diff
@@ -23,13 +34,65 @@ class SubmissionRequestProcessor(
 
         logger.info { "$accNo ${sub.owner} Started copying files accNo='${sub.accNo}', version='$version'" }
 
-        val processed = systemService.persistSubmissionFiles(sub)
+        val processed = persistSubmissionFiles(sub)
         persistenceService.expirePreviousVersions(sub.accNo)
         persistenceService.saveSubmission(processed)
-        persistenceService.saveSubmissionRequest(request.copy(status = FILES_COPIED, submission = processed))
+        val processedRequest = request.copy(status = FILES_COPIED, submission = processed, currentIndex = 0)
+        persistenceService.saveSubmissionRequest(processedRequest)
 
         logger.info { "$accNo ${sub.owner} Finished copying files accNo='$accNo', version='$version'" }
 
         return processed
     }
+
+    private fun persistSubmissionFiles(sub: ExtSubmission) = when (sub.storageMode) {
+        NFS -> persistNfsFiles(sub)
+        FIRE -> persistFireFiles(sub)
+    }
+
+    private fun persistFireFiles(sub: ExtSubmission): ExtSubmission {
+        logger.info { "${sub.accNo} ${sub.owner} Started persisting submission files on FIRE" }
+
+        val persisted = fileProcessingService.processFiles(sub) { file, idx ->
+            logger.info { "${sub.accNo} ${sub.owner} Started persisting file $idx, path='${file.filePath}' on FIRE" }
+
+            val request = FireFilePersistenceRequest(sub.accNo, sub.version, sub.relPath, file)
+            val fireFile = fireFilesService.persistSubmissionFile(request)
+            persistenceService.updateRequestIndex(sub.accNo, sub.version, idx)
+
+            logger.info { "${sub.accNo} ${sub.owner} Finished persisting file $idx, path='${file.filePath}' on FIRE" }
+
+            fireFile
+        }
+
+        logger.info { "${sub.accNo} ${sub.owner} Finished persisting submission files on FIRE" }
+
+        return persisted
+    }
+
+    private fun persistNfsFiles(sub: ExtSubmission): ExtSubmission {
+        logger.info { "${sub.accNo} ${sub.owner} Started persisting submission files on NFS" }
+
+        val permissions = sub.permissions()
+        val subFolder = nfsFilesService.getOrCreateSubmissionFolder(sub, permissions.folder)
+        val targetFolder = nfsFilesService.createTempSubFolder(subFolder, sub.accNo)
+
+        val persisted = fileProcessingService.processFiles(sub) { file, idx ->
+            logger.info { "${sub.accNo}, ${sub.owner} Started persisting file $idx, path='${file.filePath}' on NFS" }
+
+            val request = NfsFilePersistenceRequest(file as NfsFile, subFolder, targetFolder, permissions)
+            val nfsFile = nfsFilesService.persistSubmissionFile(request)
+            persistenceService.updateRequestIndex(sub.accNo, sub.version, idx)
```

Comment:
as we are not doing anything with this index I would rather include it when we actually have a use for it (also it will help to reduce this PR scope, size and delivery time).